src/libraries/System.Private.Uri/src/System/ValueStringBuilderExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace System.Text
+{
+    internal ref partial struct ValueStringBuilder
+    {
+        public void Replace(char oldChar, char newChar)
```

Comment:
This string.Replace this is replacing is vectorized; this is not.  Might that matter?

Reply:
I pushed a new commit so the call site it now replaces is only
```c#
for (ushort i = 0; i < (ushort)count; ++i)
{
    if (result[i] == '/')	
    {
        result[i] = '\\';	
    }
}
```

The `string.Replace` got replaced by
```c#
for (int i = 0; i < localPath.Length; i++)
{
    char currentChar = localPath[i];
    buffer[i] = currentChar == '/' ? '\\' : currentChar;
}
```
so there might be a perf regression in specific cases here, I will run some tests

Reply:
This can regress cases of Unc paths that do not contain any non-Ascii characters, have no compressible segments and have a forward slash relatively close to the beginning, while being long enough for vectorization to matter. For medium-length paths of this kind, the difference is ~4%.
I believe this is acceptable.

Reply:
What if you wrote it like so, what would be the perf difference?

```cs
// Pseudocode
var span = _chars.Slice(0, _pos);
while (span.Length != 0)
{
    int index = Span<char>.IndexOf(span, oldChar);

    if (index < 0) break;

    span[index] = newChar;
    span = span.Slice(index + 1);
}
```

Reply:
If you are referring to the `ValueStringBuilder.Replace`, it can be improved (at least an IndexOf to start with). I haven't measured the perf of that one as it's replacing the clearly worse
```c#
for (ushort i = 0; i < (ushort)count; ++i)
{
    if (result[i] == '/')	
    {
        result[i] = '\\';	
    }
}
```